Source/FolioReaderCenter.swift

```diff
@@ -9,120 +9,124 @@
-    var activityIndicator = UIActivityIndicatorView()
-
-    fileprivate var screenBounds: CGRect!
-    fileprivate var pointNow = CGPoint.zero
-    fileprivate var pageOffsetRate: CGFloat = 0
-    fileprivate var tempReference: FRTocReference?
-    fileprivate var isFirstLoad = true
-	fileprivate var currentWebViewScrollPositions = [Int: CGPoint]()
-	fileprivate var currentOrientation: UIInterfaceOrientation?
-    
+    open fileprivate(set) var currentPage   : FolioReaderPage?
+
+    var collectionView                      : UICollectionView!
+    let collectionViewLayout                = UICollectionViewFlowLayout()
+    var loadingView                         :UIActivityIndicatorView!
+    var pages                               : [String]!
+    var totalPages                          : Int = 0
+    var tempFragment                        : String?
+    var animator                            : ZFModalTransitionAnimator!
+    var pageIndicatorView                   : FolioReaderPageIndicator?
+    var pageIndicatorHeight                 : CGFloat = 20
+    var recentlyScrolled                    = false
+    var recentlyScrolledDelay               = 2.0 // 2 second delay until we clear recentlyScrolled
+    var recentlyScrolledTimer               : Timer!
+    var scrollScrubber                      : ScrollScrubber?
+    var activityIndicator                   = UIActivityIndicatorView()
+    var isScrolling                         = false
+    var pageScrollDirection                 = ScrollDirection()
+    var nextPageNumber                      : Int = 0
+    var previousPageNumber                  : Int = 0
+    var currentPageNumber                   : Int = 0
+    var pageWidth                           : CGFloat = 0.0
+    var pageHeight                          : CGFloat = 0.0
+
+    fileprivate var screenBounds                    : CGRect!
+    fileprivate var pointNow                        = CGPoint.zero
+    fileprivate var pageOffsetRate                  : CGFloat = 0
+    fileprivate var tempReference                   : FRTocReference?
+    fileprivate var isFirstLoad                     = true
+    fileprivate var currentWebViewScrollPositions   = [Int: CGPoint]()
+    fileprivate var currentOrientation              : UIInterfaceOrientation?
+
+    fileprivate var book : FRBook {
+        return self.readerContainer.book
+    }
+
+    fileprivate var readerConfig : FolioReaderConfig {
+        return self.readerContainer.readerConfig
+    }
+
     // MARK: - Init
-    
-    init() {
+
+    init(withContainer readerContainer: FolioReaderContainer) {
+        self.readerContainer = readerContainer
         super.init(nibName: nil, bundle: Bundle.frameworkBundle())
-        initialization()
+
+        self.initialization()
     }
-    
+
     required public init?(coder aDecoder: NSCoder) {
-        super.init(coder: aDecoder)
-        initialization()
+        fatalError("This class doesn't support NSCoding.")
```

Comment:
Because the `FolioReaderCenter` class must be initialised with non-nil `readerContainer: FolioReaderContainer` variable.

```swift
open var readerContainer: FolioReaderContainer

init(withContainer readerContainer: FolioReaderContainer) {
        self.readerContainer = readerContainer
        super.init(nibName: nil, bundle: Bundle.frameworkBundle())

        self.initialization()
}
```

Without a `readerContainer` the `FolioReaderCenter` class cannot be / should be instantiated.